kafka-persistence/src/main/scala/com/evolutiongaming/kafka/flow/kafkapersistence/KafkaPersistence.scala

```diff
@@ -0,0 +1,166 @@
+package com.evolutiongaming.kafka.flow.kafkapersistence
+
+
+import cats.data.NonEmptyList
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import cats.mtl.MonadState
+import cats.{FlatMap, Monad, data}
+import com.evolutiongaming.catshelper.{BracketThrowable, FromTry, Log}
+import com.evolutiongaming.kafka.flow.KafkaKey
+import com.evolutiongaming.kafka.flow.key.{Keys, KeysOf}
+import com.evolutiongaming.kafka.flow.persistence.{PersistenceOf, SnapshotPersistenceOf}
+import com.evolutiongaming.kafka.flow.snapshot.{SnapshotDatabase, Snapshots, SnapshotsOf}
+import com.evolutiongaming.kafka.journal.ConsRecord
+import com.evolutiongaming.kafka.journal.util.SkafkaHelper._
+import com.evolutiongaming.skafka.consumer.{ConsumerConfig, ConsumerOf, ConsumerRecord, WithSize}
+import com.evolutiongaming.skafka.producer.{Producer, ProducerRecord}
+import com.evolutiongaming.skafka.{FromBytes, Offset, Partition, ToBytes, Topic, TopicPartition}
+import com.evolutiongaming.sstream.Stream
+import com.olegpy.meow.effects._
+import scodec.bits.ByteVector
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+final case class KafkaPersistence[F[_], K, S, A](
+                                                  keysOf: KeysOf[F, K],
+                                                  snapshots: SnapshotPersistenceOf[F, K, S, A],
+                                                  onRecoveryFinished: F[Unit]
```

Comment:
It seems that `onRecoveryFinished` does an important internal job in an implementation. Should it really be exposed in API?

Reply:
Well I can say that no members of `KafkaPersistence` are supposed to be used by the application code, it may (and should) be an opaque type for the client. Thanks for the reminder, will do that.

Reply:
Done.